permissions_validators/src/lib.rs

```diff
@@ -15,83 +17,33 @@ use iroha_data_model::{isi::*, prelude::*};
 use iroha_macro::error::ErrorTryFromEnum;
 use serde::Serialize;
 
-macro_rules! impl_from_item_for_instruction_validator_box {
-    ( $ty:ty ) => {
-        impl From<$ty> for IsInstructionAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                Box::new(validator)
-            }
-        }
-    };
-}
-
-macro_rules! impl_from_item_for_query_validator_box {
-    ( $ty:ty ) => {
-        impl From<$ty> for IsQueryAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                Box::new(validator)
-            }
-        }
-    };
-}
-
-macro_rules! impl_from_item_for_granted_token_validator_box {
-    ( $ty:ty ) => {
-        impl From<$ty> for HasTokenBoxed {
-            fn from(validator: $ty) -> Self {
-                Box::new(validator)
-            }
-        }
-
-        impl From<$ty> for IsInstructionAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                let validator: HasTokenBoxed = validator.into();
-                Box::new(validator)
-            }
+macro_rules! try_evaluate_or_deny {
+    ($e:expr, $wsv:ident) => {
+        match $e.evaluate($wsv, &Context::new()) {
+            Ok(value) => value,
+            Err(err) => return ValidatorVerdict::Deny(err.to_string()),
         }
     };
 }
 
-macro_rules! impl_from_item_for_grant_instruction_validator_box {
-    ( $ty:ty ) => {
-        impl From<$ty> for IsGrantAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                Box::new(validator)
-            }
-        }
-
-        impl From<$ty> for IsInstructionAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                let validator: IsGrantAllowedBoxed = validator.into();
-                Box::new(validator)
-            }
-        }
-    };
-}
-
-macro_rules! impl_from_item_for_revoke_instruction_validator_box {
-    ( $ty:ty ) => {
-        impl From<$ty> for IsRevokeAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                Box::new(validator)
-            }
-        }
-
-        impl From<$ty> for IsInstructionAllowedBoxed {
-            fn from(validator: $ty) -> Self {
-                let validator: IsRevokeAllowedBoxed = validator.into();
-                Box::new(validator)
-            }
+/// TODO: Use [`FromResidual`](https://doc.rust-lang.org/std/ops/trait.FromResidual.html)
+/// once it becomes stable
```

Comment:
if it's a TODO, I think it should be a regular comment, not a doc one

```suggestion
// TODO: Use [`FromResidual`](https://doc.rust-lang.org/std/ops/trait.FromResidual.html)
// once it becomes stable
```